app/src/main/java/org/fossasia/openevent/general/order/OrdersUnderUserViewModel.kt

```diff
@@ -3,112 +3,103 @@ package org.fossasia.openevent.general.order
                 mutableShowShimmerResults.value = true
-                mutableNoTickets.value = false
             }.subscribe({
-                order = it
-                val eventIds = it.mapNotNull { order -> order.event?.id }
-                if (eventIds.isNotEmpty()) {
-                    eventsUnderUser(eventIds, showExpired)
+                val currentPagedOrdersAndEvents = mutableEventAndOrderPaged.value
+                if (currentPagedOrdersAndEvents == null) {
+                    mutableEventAndOrderPaged.value = it
                 } else {
-                    mutableShowShimmerResults.value = false
-                    mutableNoTickets.value = true
-                }
-            }, {
-                mutableShowShimmerResults.value = false
-                mutableNoTickets.value = true
-                mutableMessage.value = resource.getString(R.string.list_orders_fail_message)
-                Timber.d(it, "Failed  to list Orders under a user ")
-            })
-    }
-
-    private fun eventsUnderUser(eventIds: List<Long>, showExpired: Boolean) {
-        compositeDisposable += eventService.getEventsUnderUser(eventIds)
-            .withDefaultSchedulers()
-            .distinctUntilChanged()
-            .doFinally {
-                mutableShowShimmerResults.value = false
-            }.subscribe({
-                mutableShowShimmerResults.value = false
-                val events = ArrayList<Event>()
-                it.map {
-                    val times = eventIdAndTimes[it.id]
-                    if (times != null) {
-                        for (i in 0..times) {
-                            events.add(it)
-                        }
-                    }
-                    eventIdMap[it.id] = it
-                }
-                val eventAndIdentifier = ArrayList<Pair<Event, Order>>()
-                order.forEach {
-                    val event = eventIdMap[it.event?.id]
-                    if (event != null)
-                        eventAndIdentifier.add(Pair(event, it))
-                }
-                val finalList = when (showExpired) {
-                    false -> eventAndIdentifier.filter {
-                        EventUtils.getTimeInMilliSeconds(it.first.endsAt, null) > System.currentTimeMillis() }
-                    true -> eventAndIdentifier.filter {
-                        EventUtils.getTimeInMilliSeconds(it.first.endsAt, null) < System.currentTimeMillis() }
+                    currentPagedOrdersAndEvents.addAll(it)
+                    mutableEventAndOrderPaged.value = currentPagedOrdersAndEvents
                 }
-                if (finalList.isEmpty()) mutableNoTickets.value = true
-                mutableEventAndOrder.value = finalList
             }, {
-                mutableShowShimmerResults.value = false
                 mutableMessage.value = resource.getString(R.string.list_events_fail_message)
                 Timber.d(it, "Failed  to list events under a user ")
```

Comment:
I think it will never through the error, the error should handle in the order data source class.

Reply:
make sense, I'll move it to the data source class

Reply:
updated